src/objects.c

```diff
@@ -1828,8 +1824,7 @@ Obj FuncSWITCH_OBJ(Obj self, Obj obj1, Obj obj2) {
     REGION(obj2) = ds1;
     REGION(obj1) = ds2;
 #endif
-    PTR_BAG(obj2) = ptr1;
-    PTR_BAG(obj1) = ptr2;
+    SwapMasterPoint(obj1, obj2);
```

Comment:
In the non-HPCGAP case, `FuncFORCE_SWITCH_OBJ` just calls `FuncSWITCH_OBJ`. In HPCGAP, we can just rewrite `PTR_BAG`s directly (as there is no gasman book-keeping to fix).